notifications/pipeline_steps/base.py

```diff
@@ -24,5 +24,7 @@ def process_message(message):
         logging.info("%s has received %s %s", message.current_step, message.name, message.uuid)
         if pytz.utc.localize(datetime.utcnow()) > message.expiration_time:
```

Comment:
As it stands, that's true but in the future, each of the processing steps might want to drop the message if it expires at the step itself right? Rather than wait for the processing to complete return and then drop the message.